src/appmixer/utils/csv/CSVProcessor.js

```diff
@@ -401,33 +400,37 @@ module.exports = class CSVProcessor {
                 await lock.extend(lockExtendTime);
             }, config.lockExtendInterval || 10000);
 
-            await this.loadHeaders();
-            stream = await this.loadFile();
-            writeStream = new PassThrough();
-
             const rowsToAdd = this.withHeaders ? this.addHeaders(rows, this.getHeaders()) : rows;
 
             // If all the new rows are empty, warn the user.
             if (rowsToAdd.every(row => row.every(cell => !cell))) {
                 this.context.log({ warning: 'Empty rows added', details: 'Please make sure you are adding the correct data and using the correct delimiter.' });
             }
 
-            // append existing rows
-            for await (const rowData of stream) {
-                writeStream.write(this.formatRow(rowData));
-            }
-
-            // append new rows
-            this.writeRows(writeStream, rowsToAdd);
+            await this.loadHeaders();
+            readStream = await this.context.getFileReadStream(this.fileId);
+            writeStream = new PassThrough();
 
-            writeStream.end();
+            const promise = new Promise((resolve, reject) => {
+                const stream = pipeline(
+                    // (readStream + rowsToAdd) -> writeStream
+                    readStream.on('end', (foo) => {
```

Comment:
Does this even work? How do you pipe the `readStream` to the `writeStream`? I cannot see it.
Shouldn't it look something like:
<img width="705" alt="ChatGPT" src="https://github.com/user-attachments/assets/d1466a6a-f394-4f98-af86-fb4773f9eef0">

Reply:
It worked fine. But I moved the `on` part above for clarity. Also removed the unused parameter.